### PR TITLE
Windows VSTS Agent install update

### PR DESCRIPTION
**Pull request by davidrea-MS**
Updating VSTS agent install to use v2.* agents
Updating VSTS agent install to take account name and password parameters
Updating VSTS agent install to take Working Directory parameter
Updating VSTS agent install to fail on Write-Error

I can make these all independent pull requests if preferred. The last commit adjusts indentation, so looks bigger than it is.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@davidrea-MS__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: Artifacts/windows-vsts-build-agent/vsts-agent-install.ps1 [hunk 1]
```diff
@@ -18,110 +18,126 @@ Param(
...
-    if ($vstsAccount -match "https*://" -or $vstsAccount -match "visualstudio.com")
-    {
-        Write-Error "VSTS account should not be the URL, just the account name."
-        exit 1
-    }
+    Write-Error "VSTS account should not be the URL, just the account name."
+}
 
-    $currentLocation = Split-Path -parent $MyInvocation.MyCommand.Definition
+if ($workDirectory -ne "" -and !(Test-Path -Path $workDirectory -IsValid -ErrorAction Ignore)) {
```
Comment (drwill-ms) [inline Artifacts/windows-vsts-build-agent/vsts-agent-install.ps1]: Minor, brace on next line

**Comment by drwill-ms** [comment]
Have you tested this with specifying an account, and taking the default?

**Comment by davidrea-MS** [comment]
Yes, I've tested keeping the defaults for everything, and specifying everything. The windowsLogonAccount defaults to the same account as before.

File: Artifacts/windows-vsts-build-agent/vsts-agent-install.ps1 [hunk 2]
```diff
@@ -18,110 +18,128 @@ Param(
...
-    $serviceDisplayName = "VSTS Agent ($vstsAccount.$agentName)"
-    &start cmd.exe "/k $agentExePath /configure /RunningAsService /login:$vstsUser,$vstsUserPassword /serverUrl:$serverUrl ""/WindowsServiceLogonAccount:NT AUTHORITY\NetworkService"" /WindowsServiceLogonPassword /WindowsServiceDisplayName:""$serviceDisplayName"" /name:""$agentName"" /poolname:""$poolname"" /WorkFolder:$WorkFolder /StartMode:Automatic /force /NoPrompt &exit"
+# Set the current directory to the agent dedicated one previously created.
+Push-Location -Path $agentInstallationPath
+# The actual install of the agent. Using --runasservice, and some other values that could be turned into paramenters if needed.
+$agentConfigArgs = "--unattended", "--url", $serverUrl, "--auth", "PAT", "--token", $vstsUserPassword, "--pool", $poolname, "--agent", $agentName, "--runasservice", "--windowslogonaccount", $windowsLogonAccount
+if ($windowsLogonPassword -ne "") {
+    $agentConfigArgs += "--windowslogonpassword", $windowsLogonPassword
+}
+if ($workDirectory -ne "") {
```
Comment (drwill-ms) [inline Artifacts/windows-vsts-build-agent/vsts-agent-install.ps1]: braces here too. and line 138

**Comment by drwill-ms** [comment]
Ignored CLA required because I know you are with MS, and was able to join the org. Looks like the tool doesn't recognize that, and I wouldn't make you resubmit.

**Pull request merged**